tools/list_artifacts.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Tool to list artifacts by operating system."""
+
+import argparse
+import csv
+import os
+import sys
+
+from artifacts import definitions
+from artifacts import reader
+
+
+class ArtifactLister():
+  """Generate a listing of artifact definitions for an operating system.
+
+  Attributes:
+    directory (str): the data directory of artifact definitions.
+    target_os (str): the target operating system.
+    split_paths (bool): split artifict definitions into separate entries when
+       there are more than one path values.
+  """
+
+  def __init__(self, directory, target_os, split_paths=False):
+    """Initializes an artifact lister.
+
+    Args:
+      directory: the data directory of artifact definitions.
+      target_os: the target operating system.
+      split_paths: split artifict definitions into separate entries when
+         there are more than one path values."""
+    self.directory = directory
+    self.target_os = target_os
+    self.split_paths = split_paths
+
+  def _LoadArtifacts(self):
+    """Generates a dictionary representing an artifact."""
```

Comment:
add Yields information to docstring.

Why not break this large function up in smaller functions for the different type of source types ?

Reply:
Done